src/app/context/btcWallet/BtcWallet.tsx

```diff
@@ -0,0 +1,56 @@
+import { networks } from "bitcoinjs-lib";
+import React, { createContext, ReactNode, useContext, useState } from "react";
+
+import { WalletProvider } from "@/utils/wallet/wallet_provider";
+interface BtcWalletContextType {
+  publicKeyNoCoord: string;
+  address: string;
+  btcWallet: WalletProvider | undefined;
+  btcWalletNetwork: networks.Network | undefined;
+  setPublicKeyNoCoord: (key: string) => void;
+  setAddress: (address: string) => void;
+  setBtcWallet: (wallet: WalletProvider | undefined) => void;
+  setBtcWalletNetwork: (network: networks.Network | undefined) => void;
+}
+
+const BtcWalletContext = createContext<BtcWalletContextType | undefined>(
+  undefined,
+);
+
+export const BtcWalletProvider: React.FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  const [publicKeyNoCoord, setPublicKeyNoCoord] = useState("");
+  const [address, setAddress] = useState("");
+  const [btcWallet, setBtcWallet] = useState<WalletProvider | undefined>(
+    undefined,
+  );
+  const [btcWalletNetwork, setBtcWalletNetwork] = useState<
+    networks.Network | undefined
+  >(undefined);
+
+  const value = {
+    publicKeyNoCoord,
+    address,
+    btcWallet,
+    btcWalletNetwork,
+    setPublicKeyNoCoord,
+    setAddress,
+    setBtcWallet,
+    setBtcWalletNetwork,
+  };
+
+  return (
+    <BtcWalletContext.Provider value={value}>
+      {children}
+    </BtcWalletContext.Provider>
+  );
+};
+
+export const useBtcWallet = () => {
```

Comment:
The introduction of this seems unrelated to the PR. Why did we add it here instead of a separate PR that can be reviewed independently?